bin/s2nd.c

```diff
@@ -748,13 +748,22 @@ int main(int argc, char *const *argv)
         }
     }
 
+    if (parallelize) {
+        struct sigaction sa;
+
+        sa.sa_handler = SIG_IGN;
+        sa.sa_flags = SA_NOCLDWAIT;
+        sigemptyset(&sa.sa_mask);
+        sigaction(SIGCHLD, &sa, NULL);
+    }
+
     int fd;
     while ((fd = accept(sockfd, ai->ai_addr, &ai->ai_addrlen)) > 0) {
 
         if (!parallelize) {
             int rc = handle_connection(fd, config, conn_settings);
-            close(fd);
             if (rc < 0) {
+            close(fd);
```

Comment:
nit: spacing

Also can we move the responsibility to close an fd fully into `handle_connection` or fully our of it. Having it in 2 places is just asking for errors :D

Reply:
let's move it out :)  makes sense.